src/main/java/nextstep/subway/line/domain/Sections.java

```diff
@@ -7,25 +7,132 @@
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
+import java.util.Objects;
+import java.util.function.Predicate;
 
 import javax.persistence.Embeddable;
 import javax.persistence.OneToMany;
 
+import nextstep.subway.common.exception.DuplicationException;
+import nextstep.subway.common.exception.ServiceException;
+import nextstep.subway.line.exception.SectionNotFoundException;
+import nextstep.subway.station.domain.Station;
+
 @Embeddable
 public class Sections {
 
+
+    public static final String MESSAGE_IS_NOT_EXISTS_STATION = "기존 노선에 해당역이 존재하지 않습니다. 상행역[%s] 하행역[%s]";
+
     @OneToMany(mappedBy = "line", fetch = LAZY, cascade = ALL)
     private List<Section> sections = new ArrayList<>();
 
-    public void add(List<Section> sections, Line line) {
-        this.sections = sections.stream()
-                                .peek(section -> section.setLine(line))
-                                .collect(toList());
+    public void add(List<Section> newSections, Line line) {
+        List<Section> addNewSections = newSections.stream()
+                                                  .peek(section -> section.setLine(line))
+                                                  .collect(toList());
+        this.sections.addAll(addNewSections);
+    }
+
+    public void add(Section newSection, Line line) {
+        if (sections.contains(newSection)) {
+            return;
+        }
+        newSection.setLine(line);
+        sections.add(newSection);
+    }
+
+    public void addSection(Section newSection) {
+
+        if (matchStation(isDownStation(newSection))) {
+            addUpSectionOrMiddle(newSection);
+            return;
+        }
+
+        if (matchStation(isUpStation(newSection))) {
+            addDownSection(newSection);
+            return;
+        }
+        throw new ServiceException(String.format(MESSAGE_IS_NOT_EXISTS_STATION, newSection.getUpStation(), newSection.getDownStation()));
+    }
+
+    private boolean matchStation(Predicate<Section> isStation) {
+        return sections.stream()
+                       .anyMatch(isStation);
+    }
+
+    private Predicate<Section> isDownStation(Section newSection) {
+        return section -> section.findUpStation(newSection);
```

Comment:
Boolean을 반환하기 때문에 find라는 메서드명 보다는 is로 변경하는 것이 좋을 것 같습니다!

```suggestion
        return section -> section.isUpStation(newSection);
```